submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/FireService.kt

```diff
@@ -1,64 +1,81 @@
 package ac.uk.ebi.biostd.persistence.filesystem.fire
 
+import ac.uk.ebi.biostd.persistence.filesystem.extensions.fireType
 import ebi.ac.uk.extended.model.ExtFile
-import ebi.ac.uk.extended.model.ExtFileType
+import ebi.ac.uk.extended.model.ExtFileType.DIR
+import ebi.ac.uk.extended.model.ExtFileType.FILE
 import ebi.ac.uk.extended.model.ExtSubmission
 import ebi.ac.uk.extended.model.FireFile
 import ebi.ac.uk.extended.model.NfsFile
-import ebi.ac.uk.io.ext.md5
 import mu.KotlinLogging
 import org.zeroturnaround.zip.ZipUtil
 import uk.ac.ebi.fire.client.integration.web.FireClient
-import uk.ac.ebi.fire.client.model.FireApiFile
+import uk.ac.ebi.fire.client.model.isAvailable
+import uk.ac.ebi.fire.client.model.path
 import java.io.File
 import java.nio.file.Files
 
 private val logger = KotlinLogging.logger {}
 
 class FireService(
-    private val fireClient: FireClient,
+    private val client: FireClient,
     private val fireTempDirPath: File,
 ) {
-    fun getOrPersist(sub: ExtSubmission, file: ExtFile): FireFile {
-        return when (file) {
-            is FireFile -> reuseFireFile(sub, file)
-            is NfsFile -> getOrPersist(sub, file)
-        }
-    }
 
     fun cleanFtp(sub: ExtSubmission) {
-        fireClient
+        logger.info { "Clean FTP files (unset path) of files in submission ${sub.accNo}" }
+        client
             .findByAccNo(sub.accNo)
-            .forEach { fireClient.unsetPath(it.fireOid) }
+            .apply { logger.info { "Found $size files in submission ${sub.accNo}" } }
+            .forEach { client.unsetPath(it.fireOid) }
+        logger.info { "Finish clean FTP files (unset path) of files in submission ${sub.accNo}" }
     }
 
-    private fun reuseFireFile(sub: ExtSubmission, fireFile: FireFile): FireFile {
-        val newPath = "${sub.relPath}/${fireFile.relPath}"
-        fireClient.setPath(fireFile.fireId, newPath)
-        return fireFile
+    /**
+     * Get or persist the given ext file. Note that this method assumes all previous submission versions  fire files has
```

Comment:
A couple of typos mixed with a couple of suggestions:

Get or persist the given ext file from FIRE. Note that this method assumes that all the fire files belonging to previous submission versions have been unpublished and any file with a path is assumed to be already used.

For both FIRE and NFS, the file is searched by md5, and the system checks that it does not belong to another submission. The method also ensures that the file has no path (i.e. it was submitted in the same submission in a different path) and if so, even if the file exists in FIRE, it gets duplicated to ensure consistency.